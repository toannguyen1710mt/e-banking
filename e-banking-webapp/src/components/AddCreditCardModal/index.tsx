'use client';

import { useEffect, useState, useTransition } from 'react';
import Image from 'next/image';
import { z } from 'zod';
import { zodResolver } from '@hookform/resolvers/zod';
import { useForm } from 'react-hook-form';
import { AuthError, Session } from 'next-auth';

// Constants
import { ERROR_MESSAGES, IMAGES } from '@/constants';

// Interfaces
import { IAccount } from '@/interfaces';

// Actions
import { addNewCardByAccountId } from '@/actions';

// Schemas
import { CreditCardSchema } from '@/schemas';

// Services
import { getAccountsByUserId } from '@/services';

// Components
import * as WizardForm from '@/components/common/WizardForm';
import { Modal } from '../common';
import { AddCreditCard } from './AddCreditCard';
import { ConfirmAddCard } from './ConfirmAddCard';
import { toastStore } from '@/utils';

interface IAddCreditCardModalProps {
  session: Session;
  isOpen: boolean;
  onClose: () => void;
}

type FormValues = z.infer<typeof CreditCardSchema>;

export const AddCreditCardModal = ({
  isOpen,
  session,
  onClose,
}: IAddCreditCardModalProps) => {
  const [accounts, setAccounts] = useState<IAccount[]>([]);
  const [isPending, startTransition] = useTransition();

  const form = useForm<FormValues>({
    resolver: zodResolver(CreditCardSchema),
    defaultValues: {
      cardInfo: {
        holderName: '',
        cardNumber: '',
        expireAt: '',
        ccv: '',
      },
      walletType: 'Main',
    },
    reValidateMode: 'onBlur',
    mode: 'onBlur',
  });

  useEffect(() => {
    const fetchData = async () => {
      try {
        const result = await getAccountsByUserId(session.user.id);

        setAccounts(result || []);
      } catch (error) {
        if (error instanceof AuthError) {
          throw ERROR_MESSAGES.GET_ERROR;
        }
      }
    };

    fetchData();
  }, [session.user.id]);

  const onSubmit = async (data: FormValues) => {
    const { holderName, cardNumber, expireAt, ccv } = data.cardInfo;

    const selectedAccount = accounts.find(
      (account) => account.type === data?.walletType,
    );

    try {
      await addNewCardByAccountId(selectedAccount?.documentId as string, {
        holderName,
        cardNumber,
        expireAt,
        ccv,
      });

      startTransition(() => {
        toastStore.showToast(
          ERROR_MESSAGES.ADD_CARD_SUCCESS,
          'success',
          'top-center',
        );

        return onClose();
      });
    } catch (error) {
      toastStore.showToast(
        ERROR_MESSAGES.ADD_CARD_FAILED,
        'error',
        'top-center',
      );

      if (error instanceof AuthError) {
        throw ERROR_MESSAGES.ADD_CARD_FAILED;
      }
    }
  };

  return (
    <Modal
      isOpen={isOpen}
      onClose={onClose}
      size='4xl'
      placement='center'
      classNames={{
        base: 'p-0 rounded-xl bg-primary-200',
      }}
    >
      <div className='flex items-center justify-between'>
        <Image
          className='max-w-[50%] flex-1 px-[31px] py-[87px]'
          src={IMAGES.ADD_CREDIT_CARD_IMAGE}
          width={333}
          height={280}
          alt='Add Credit Card'
        />

        <div className='min-h-[500px] flex-1 rounded-xl bg-white px-4 py-8'>
          <WizardForm.Root
            schema={CreditCardSchema}
            form={form}
            className='flex grow flex-col'
            onSubmit={onSubmit}
          >
            <WizardForm.Step name='cardInfo' key='cardInfo'>
              <AddCreditCard />
            </WizardForm.Step>

            <WizardForm.Step
              name='confirmationDetails'
              key='confirmationDetails'
            >
              <ConfirmAddCard accounts={accounts} isPending={isPending} />
            </WizardForm.Step>
          </WizardForm.Root>
        </div>
      </div>
    </Modal>
  );
};
