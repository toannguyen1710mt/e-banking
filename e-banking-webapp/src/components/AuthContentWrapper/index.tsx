'use client';

import { ReactNode } from 'react';
import { usePathname } from 'next/navigation';
import Link from 'next/link';
import Image from 'next/image';

// Constants
import { IMAGES, ROUTES } from '@/constants';

// Interfaces
import { TEXT_SIZE, TEXT_VARIANT } from '@/interfaces';

// Components
import { Text } from '../common';

interface AuthContentWrapperProps {
  textFooter?: string;
  textHeading: ReactNode;
  formContent: ReactNode;
  subheading?: string;
}

export const AuthContentWrapper = ({
  textHeading,
  formContent,
  textFooter,
  subheading,
}: AuthContentWrapperProps) => {
  const pathname = usePathname();

  const isOnSignIn = pathname === ROUTES.SIGN_IN;

  const linkDetails = {
    text: isOnSignIn ? 'Register' : 'Sign In',
    href: isOnSignIn ? ROUTES.SIGN_UP : ROUTES.SIGN_IN,
  };

  const classLogo = isOnSignIn
    ? 'mt-0 md:mt-[108px] pb-[56px] md:pb-9'
    : 'pb-[14px] lg:pb-9';
  const classTextHeading = isOnSignIn ? 'mb-7' : 'mb-0 md:mb-7';

  return (
    <div className='mx-auto flex flex-1 flex-col'>
      {/* Logo */}
      <Image
        className={`${classLogo}`}
        src={IMAGES.LOGO}
        width={104}
        height={36}
        alt='Logo EBanking'
      />

      <div className='max-w-[406px] text-center md:text-left'>
        <Text
          as='h1'
          size={TEXT_SIZE['4XL']}
          variant={TEXT_VARIANT.DEFAULT}
          className={`${classTextHeading}`}
        >
          {textHeading}
        </Text>
        <Text
          size={TEXT_SIZE.BASE}
          variant={TEXT_VARIANT.INFO}
          className='mb-4 font-normal leading-[22px]'
        >
          {subheading}
        </Text>
      </div>

      {/* Form */}
      <div className='mx-auto mb-[17px] max-w-[325px] md:mx-0'>
        {formContent}
      </div>

      {textFooter && (
        <Text
          size={TEXT_SIZE.SM}
          variant={TEXT_VARIANT.DEFAULT}
          className='text-center leading-[20px] md:text-left'
        >
          {textFooter}&nbsp;
          <Link
            href={linkDetails.href}
            className='leading-[20px] text-primary-200'
          >
            {linkDetails.text}
          </Link>
        </Text>
      )}
    </div>
  );
};
