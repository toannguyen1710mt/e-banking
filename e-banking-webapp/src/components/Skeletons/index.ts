export * from './SkeletonTable';
export * from './CreditCardSkeleton';
export * from './AnalyticsCardSkeleton';
export * from './ChartsSkeleton';
export * from './TransactionHistorySkeleton';
