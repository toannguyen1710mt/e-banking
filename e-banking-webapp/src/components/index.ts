export * from './common';
export * from './icons';
export * from './Error';
export * from './NotFound';
export * from './Slideshow';
export * from './AuthContentWrapper';
export * from './LoginForm';
export * from './TargetTile';
export * from './QuickAction';
export * from './SignUpForm';
export * from './Charts';
export * from './Sidebar';
export * from './CreditCard';
export * from './Transfers';
export * from './ExpenseAnalysis';
export * from './StepProgress';
export * from './MyCards';
export * from './BalanceModal';
export * from './MyTarget';
export * from './SettingContainer/Account/ConnectedAccountsTab';
export * from './SettingContainer/General';
export * from './SettingContainer/Account/DeleteAccountTab';
export * from './ServiceCard';
export * from './SettingContainer/Account';
export * from './DueTile';
export * from './MasterCard';
export * from './CardOverview';
export * from './MetricsCard';
export * from './Skeletons';
export * from './TasksWithCalendar';
export * from './ContainerAnalytics';
export * from './AddCreditCardModal';
export * from './InformationCard';
export * from './AddCreditCardModal/AddCreditCard';
export * from './AnalyticsCard';
export * from './DailyLimit';
export * from './TransactionTable';
export * from './TransactionHistory';
export * from './ContainerTransactions';
export * from './MonthYearPicker';
