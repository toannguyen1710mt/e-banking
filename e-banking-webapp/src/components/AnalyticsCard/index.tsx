'use client';

// Libs
import { Card, CardBody, CardFooter, CardHeader } from '@nextui-org/react';

// Utils
import { formatNumberWithCommas } from '@/utils';

// Components
import { ArrowDownIcon, ArrowUpIcon, Chip, Text } from '@/components';

export interface IAnalyticsCardItem {
  title: string;
  subtitle: string;
  amount: number;
  percentageChange: number;
  isPositive?: boolean;
}

export const AnalyticsCard = ({
  title,
  subtitle,
  isPositive,
  amount,
  percentageChange,
}: IAnalyticsCardItem) => {
  const ArrowComponent = isPositive ? <ArrowUpIcon /> : <ArrowDownIcon />;

  return (
    <Card key={title} className='w-full gap-4 rounded-md p-0 shadow-none'>
      <CardHeader className='p-4 text-xs font-medium'>
        <Text>{title}</Text>
      </CardHeader>
      <CardBody className='flex flex-row items-center gap-5 overflow-y-hidden p-4'>
        <Text as='span' className='text-[2em] font-extrabold'>
          ${formatNumberWithCommas(amount)}
        </Text>
        <Chip
          startContent={ArrowComponent}
          color={isPositive ? 'success' : 'danger'}
          classNames={{
            base: 'h-5',
            content: 'font-extrabold text-4xs p-0',
          }}
        >
          {percentageChange}%
        </Chip>
      </CardBody>
      <CardFooter className='rounded-none p-4'>
        <Text className='text-2xs font-light text-transparentBlack'>
          {subtitle}
        </Text>
      </CardFooter>
    </Card>
  );
};
