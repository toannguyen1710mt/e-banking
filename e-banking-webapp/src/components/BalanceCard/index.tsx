import { Card } from '@nextui-org/react';

// Utils
import { formatNumberWithCommas } from '@/utils';

// Interfaces
import { TEXT_SIZE, TEXT_VARIANT } from '@/interfaces';

// Components
import { Text, ArrowUpRightIcon, PaymentIcon } from '@/components';

interface BalanceCardProps {
  amount: number;
  title: string;
}

export const BalanceCard = ({ amount, title }: BalanceCardProps) => (
  <Card className='h-[155px] w-full pl-[14px] pr-4 pt-[13px]'>
    <div className='mb-10 flex justify-between text-primary-200'>
      <PaymentIcon />
      <ArrowUpRightIcon />
    </div>

    <div className='flex flex-col gap-3'>
      <Text className='text-2xl font-extrabold text-primary-200' as='span'>
        ${formatNumberWithCommas(amount)}
      </Text>

      <Text size={TEXT_SIZE['2XS']} variant={TEXT_VARIANT.INFO}>
        {title}
      </Text>
    </div>
  </Card>
);
