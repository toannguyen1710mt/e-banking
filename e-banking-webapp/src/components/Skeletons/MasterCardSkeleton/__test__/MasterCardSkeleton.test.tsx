// Libs
import { render } from '@testing-library/react';

// Components
import { MasterCardSkeleton } from '@/components';

describe('MasterCardSkeleton component', () => {
  test('should match snapshot', () => {
    const container = render(<MasterCardSkeleton />);

    expect(container).toMatchSnapshot();
  });
});
