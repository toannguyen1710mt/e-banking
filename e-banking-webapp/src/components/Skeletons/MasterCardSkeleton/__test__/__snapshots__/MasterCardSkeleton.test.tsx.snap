// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MasterCardSkeleton component should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col relative overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none w-full min-w-[100px] bg-background-700 px-4 pb-5 pt-[14px] md:min-w-[294px]"
        tabindex="-1"
      >
        <div
          class="flex z-10 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large p-0"
        >
          <div
            class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-5 w-44 rounded-lg"
          >
            <div
              class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
            />
          </div>
        </div>
        <div
          class="relative flex w-full p-3 flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased gap-2 px-0 pb-0 pt-3"
        >
          <div
            class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-4 w-24 rounded-lg"
          >
            <div
              class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
            />
          </div>
          <div
            class="flex gap-6 text-[10px] font-normal text-white"
          >
            <div
              class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-3 w-14 rounded-lg"
            >
              <div
                class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
              />
            </div>
            <div
              class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-3 w-14 rounded-lg"
            >
              <div
                class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
              />
            </div>
          </div>
        </div>
        <div
          class="p-3 h-auto flex w-full overflow-hidden color-inherit subpixel-antialiased rounded-b-large flex-col items-start px-0 pb-0 pt-4"
        >
          <div
            class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-4 w-20 rounded-lg"
          >
            <div
              class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
            />
          </div>
          <div
            class="md:max-0 mx-auto mt-2 flex max-w-[220px] flex-row items-center gap-9 md:max-w-full"
          >
            <div
              class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-24 w-24 rounded-full"
            >
              <div
                class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
              />
            </div>
            <div
              class="flex flex-col gap-3"
            >
              <div
                class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-3 w-14 rounded-lg"
              >
                <div
                  class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
                />
              </div>
              <div
                class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-3 w-14 rounded-lg"
              >
                <div
                  class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
                />
              </div>
              <div
                class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-3 w-14 rounded-lg"
              >
                <div
                  class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col relative overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none w-full min-w-[100px] bg-background-700 px-4 pb-5 pt-[14px] md:min-w-[294px]"
      tabindex="-1"
    >
      <div
        class="flex z-10 w-full justify-start items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large p-0"
      >
        <div
          class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-5 w-44 rounded-lg"
        >
          <div
            class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
          />
        </div>
      </div>
      <div
        class="relative flex w-full p-3 flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased gap-2 px-0 pb-0 pt-3"
      >
        <div
          class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-4 w-24 rounded-lg"
        >
          <div
            class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
          />
        </div>
        <div
          class="flex gap-6 text-[10px] font-normal text-white"
        >
          <div
            class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-3 w-14 rounded-lg"
          >
            <div
              class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
            />
          </div>
          <div
            class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-3 w-14 rounded-lg"
          >
            <div
              class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
            />
          </div>
        </div>
      </div>
      <div
        class="p-3 h-auto flex w-full overflow-hidden color-inherit subpixel-antialiased rounded-b-large flex-col items-start px-0 pb-0 pt-4"
      >
        <div
          class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-4 w-20 rounded-lg"
        >
          <div
            class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
          />
        </div>
        <div
          class="md:max-0 mx-auto mt-2 flex max-w-[220px] flex-row items-center gap-9 md:max-w-full"
        >
          <div
            class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-24 w-24 rounded-full"
          >
            <div
              class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
            />
          </div>
          <div
            class="flex flex-col gap-3"
          >
            <div
              class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-3 w-14 rounded-lg"
            >
              <div
                class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
              />
            </div>
            <div
              class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-3 w-14 rounded-lg"
            >
              <div
                class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
              />
            </div>
            <div
              class="group relative overflow-hidden bg-content3 dark:bg-content2 pointer-events-none before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 data-[loaded=true]:pointer-events-auto data-[loaded=true]:overflow-visible data-[loaded=true]:!bg-transparent data-[loaded=true]:before:opacity-0 data-[loaded=true]:before:-z-10 data-[loaded=true]:before:animate-none data-[loaded=true]:after:opacity-0 transition-background !duration-300 h-3 w-14 rounded-lg"
            >
              <div
                class="opacity-0 group-data-[loaded=true]:opacity-100 transition-opacity motion-reduce:transition-none !duration-300"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
